plugins/experimental/memory_profile/memory_profile.cc

```diff
@@ -49,26 +49,28 @@ CallbackHandler(TSCont cont, TSEvent id, void *data)
     TSDebug(PLUGIN_NAME, "Message to '%s' - %zu bytes of data", msg->tag, msg->data_size);
     if (strcmp(PLUGIN_NAME, msg->tag) == 0) { // Message is for us
 #if TS_HAS_JEMALLOC
-      int retval = 0;
-      if (strncmp((char *)msg->data, "dump", msg->data_size) == 0) {
-        if ((retval = mallctl("prof.dump", nullptr, nullptr, nullptr, 0)) != 0) {
-          TSError("mallct(prof.dump) failed retval=%d errno=%d", retval, errno);
+      if (msg->data_size) {
```

Comment:
Is the size check needed here, or just in the debug message at line 72? `strncmp` should handle a zero size.

Reply:
strncmp:
>  The behavior is undefined when either lhs or rhs is the null pointer.

Not sure if the size = 0 will stop the function to check the ptr. Probably is safe this way.